github/actions/multi_client.go

```diff
@@ -77,34 +63,6 @@ func (m *multiClient) GetClientFor(ctx context.Context, githubConfigURL string,
 		return nil, fmt.Errorf("both PAT and GitHub App credentials provided. should only provide one")
 	}
 
-	key := ActionsClientKey{
-		ActionsURL: parsedGitHubURL.String(),
-		Namespace:  namespace,
-	}
-
-	if creds.AppCreds != nil {
-		key.Auth = ActionsAuth{
-			AppCreds: creds.AppCreds,
-		}
-	}
-
-	if creds.Token != "" {
-		key.Auth = ActionsAuth{
-			Token: creds.Token,
-		}
-	}
-
-	m.mu.Lock()
-	defer m.mu.Unlock()
-
-	clientWrapper, has := m.clients[key]
-	if has {
-		m.logger.Info("using cache client", "githubConfigURL", githubConfigURL, "namespace", namespace)
-		return clientWrapper.client, nil
-	}
-
-	m.logger.Info("creating new client", "githubConfigURL", githubConfigURL, "namespace", namespace)
-
 	client, err := NewClient(
```

Comment:
should we only create a new client on a cache miss?

Reply:
Then the identity of a client should be moved to some other entity which might not be ideal...
The client should know how to identify itself, and since NewClient does not issue any other calls when instantiated, it should not be expensive anymore

Reply:
We use a client.Identifier now as a caching key so we need the "potential" new client to understand if we have one like it already in the cache, but creating a new client is "free" now (aka no network calls) until you use to make some calls.